core/domain/exp_domain.py

```diff
@@ -2817,6 +2843,21 @@ def _convert_v32_dict_to_v33_dict(cls, exploration_dict):
 
         return exploration_dict
 
+    @classmethod
+    def _convert_v33_dict_to_v34_dict(cls, exploration_dict):
+        """Converts a v33 exploration dict into a v34 exploration dict."""
+        exploration_dict['schema_version'] = 34
+
+        # TODO
+        # exploration_dict['image_id_counter'] = get_number_of_images_in_exploration()
+        exploration_dict['image_id_counter'] = 0
+
+        exploration_dict['states'] = cls._convert_states_v28_dict_to_v29_dict(
+            exploration_dict['states'])
+        exploration_dict['states_schema_version'] = 29
+
```

Comment:
Note we have to update the image_counter after doing state migration.

Reply:
done